### ISSUE TITLE
No mixin named md-core

### ISSUE THREAD

**Issue by hashemnik** [eirslett/frontend-maven-plugin#571]
Hi everybody
I have problem using this plugin.

```
ERROR in ./~/css-loader!./~/postcss-loader!./~/sass-loader!./src/app-theme.scss
Module build failed:
undefined
        ^
      No mixin named md-core

Backtrace:
        stdin:6
      in D:\Projects\frontend-webapp\src\app-theme.scss (line 6, column 10)
 @ ./src/app-theme.scss 4:14-162
 @ multi ./src/styles.css ./src/app-theme.scss

ERROR in ./src/app-theme.scss
Module build failed: ModuleBuildError: Module build failed:
undefined
        ^
      No mixin named md-core

Backtrace:
        stdin:6
      in D:\Projects\frontend-webapp\src\app-theme.scss (line 6, column 10)
    at D:\Projects\frontend-webapp\node_modules\webpack\lib\NormalModule.js:141:35
    at D:\Projects\frontend-webapp\node_modules\loader-runner\lib\LoaderRunner.js:364:11
    at D:\Projects\frontend-webapp\node_modules\loader-runner\lib\LoaderRunner.js:230:18
    at context.callback (D:\Projects\frontend-webapp\node_modules\loader-runner\lib\LoaderRunner.js:111:13)
    at Object.onRender (D:\Projects\frontend-webapp\node_modules\sass-loader\index.js:273:13)
    at Object.<anonymous> (D:\Projects\frontend-webapp\node_modules\async\dist\async.js:2234:31)
    at apply (D:\Projects\frontend-webapp\node_modules\async\dist\async.js:20:25)
    at Object.<anonymous> (D:\Projects\frontend-webapp\node_modules\async\dist\async.js:56:12)
    at Object.callback (D:\Projects\frontend-webapp\node_modules\async\dist\async.js:840:16)
    at options.error (D:\Projects\frontend-webapp\node_modules\node-sass\lib\index.js:294:32)

ERROR in D:\Projects\frontend-webapp\node_modules\extract-text-webpack-plugin\loader.js?{"omit":1,"remove":false,"publicPath":""}!D:\Projects\frontend-webapp\node_modules\style-loader\index.js!D:\Projects\frontend-webapp\node_modules\css-loader\index.js!D:\Projects\frontend-webapp\node_modules\postcss-loader\index.js!D:\Projects\frontend-webapp\node_modules\sass-loader\index.js!D:\Projects\frontend-webapp\src\app-theme.scss doesn't export content
```

any body have solution for me ?

**Comment by shteeven** [comment]
I have the same issue. package.json: 
```javascript
{
  "name": "somename",
  "version": "0.0.0",
  "license": "MIT",
  "angular-cli": {},
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "^2.4.8",
    "@angular/compiler": "^2.4.8",
    "@angular/core": "^2.4.8",
    "@angular/flex-layout": "^2.0.0-beta.5",
    "@angular/forms": "^2.4.8",
    "@angular/http": "^2.4.8",
    "@angular/material": "^2.0.0-beta.2",
    "@angular/platform-browser": "^2.4.8",
    "@angular/platform-browser-dynamic": "^2.4.8",
    "@angular/router": "^3.4.8",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "rxjs": "^5.1.1",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.7.7"
  },
  "devDependencies": {
    "@angular/cli": "1.0.0-beta.32.3",
    "@angular/language-service": "^2.4.8",
    "@angular/compiler-cli": "^2.4.8",
    "@types/hammerjs": "^2.0.8",
    "@types/jasmine": "2.5.43",
    "@types/node": "^7.0.5",
    "codelyzer": "~2.0.1",
    "jasmine-core": "2.5.2",
    "jasmine-spec-reporter": "3.2.0",
    "karma": "1.5.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-coverage-istanbul-reporter": "^0.2.3",
    "karma-jasmine": "^1.0.2",
    "protractor": "~5.1.0",
    "ts-node": "2.1.0",
    "tslint": "^4.4.2",
    "typescript": "~2.1.6"
  }
}
```

And the error: 
```
ERROR in ./~/css-loader?{"sourceMap":false,"importLoaders":1}!./~/postcss-loader!./~/sass-loader!./src/custom_theme.scss
Module build failed: 
@include md-core();
        ^
      No mixin named md-core

Backtrace:
        stdin:6
      in /myPath/client/src/custom_theme.scss (line 6, column 10)
 @ ./src/custom_theme.scss 4:14-207
 @ multi ./src/styles.css ./src/custom_theme.scss

```

**Comment by hashemnik** [comment]
Hi 
Finally I found the solution.
Problem is about using material version above beta 1.
If you use beta 2 you get this error.
Use this instead:  
@angular/material": "2.0.0-beta.1",

Cheers
Hashem

**Comment by alexrun** [comment]
Hello @hashemnik ,

Do you know which changes are making this error to occur when you use material version above beta 1?

**Comment by alexrun** [comment]
I will answer myself. This is the changelog https://github.com/angular/material2/blob/master/CHANGELOG.md

**Comment by hashemnik** [comment]
@alexrun
Thanks 
Very good .

**Comment by Mintenker** [comment]
@alexrun is there any other alternative of fixing this except downgrading to @angular/material": "2.0.0-beta.1" as @hashemnik suggested ? 
Or, let me rephrase that, is it possible to use beta.2 and create custom material theme without using @include md-core(); line which throws this error ?

**Comment by hashemnik** [comment]
Hi 
Also you can update angular cli and other lib and use mat-core instead of md-core.

**Comment by alexrun** [comment]
Exactly, that's actually what I did...

**Comment by abansal77** [comment]
Here is my package.json file

`{
  "name": "glam",
  "version": "0.0.0",
  "license": "MIT",
  "angular-cli": {},
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "lint": "tslint \"src/**/*.ts\"",
    "test": "ng test",
    "pree2e": "webdriver-manager update --standalone false --gecko false",
    "e2e": "protractor"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "^2.3.1",
    "@angular/compiler": "^2.3.1",
    "@angular/core": "^2.3.1",
    "@angular/flex-layout": "^2.0.0-beta.1",
    "@angular/forms": "^2.3.1",
    "@angular/http": "^2.3.1",
    "@angular/material": "^2.0.0-beta.1",
    "@angular/platform-browser": "^2.3.1",
    "@angular/platform-browser-dynamic": "^2.3.1",
    "@angular/router": "^3.3.1",
    "@angular2-material/core": "^2.0.0-alpha.8-2",
    "angular-2-local-storage": "^1.0.0",
    "angular2-onsenui": "^1.0.0-rc.4",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "ng2-material": "^0.8.1",
    "ng2-page-scroll": "^4.0.0-beta.2",
    "onsenui": "^2.0.5",
    "primeng": "^1.1.2",
    "rxjs": "^5.0.1",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.7.2"
  },
  "devDependencies": {
    "@angular/compiler-cli": "^2.3.1",
    "@types/jasmine": "2.5.38",
    "@types/node": "^6.0.42",
    "angular-cli": "1.0.0-beta.24",
    "codelyzer": "~2.0.0-beta.1",
    "jasmine-core": "2.5.2",
    "jasmine-spec-reporter": "2.5.0",
    "karma": "1.2.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-jasmine": "^1.0.2",
    "karma-remap-istanbul": "^0.2.1",
    "protractor": "~4.0.13",
    "ts-node": "1.2.1",
    "tslint": "^4.0.2",
    "typescript": "~2.0.3"
  }
}`

I am facing the exact same issue. It is a fresh ubuntu instance with evrything installed right now. So don't think this could be version issue?

What is the possible solution?

**Comment by eirslett** [comment]
This is not an issue with this maven plugin, looks like it's an issue with Angular. So the issue and discussion shoudl be moved over there.

**Comment by zamarronstein** [comment]
only change all md with mat, this is my scss file:

@import '../node_modules/@angular/material/core/theming/all-theme';
// Plus imports for other components in your app.

// Include the base styles for Angular Material core. We include this here so that you only
// have to load a single css file for Angular Material in your app.
@include mat-core();

// Define the palettes for your theme using the Material Design palettes available in palette.scss
// (imported above). For each palette, you can optionally specify a default, lighter, and darker
// hue.
$candy-app-primary: mat-palette($mat-blue);
$candy-app-accent:  mat-palette($mat-light-blue, A200, A100, A400);

// The warn palette is optional (defaults to red).
$candy-app-warn:    mat-palette($mat-red);

// Create the theme object (a Sass map containing all of the palettes).
$candy-app-theme: mat-light-theme($candy-app-primary, $candy-app-accent, $candy-app-warn);

// Include theme styles for core and each component used in your app.
// Alternatively, you can import and @include the theme mixins for each component
// that you are using.
@include angular-material-theme($candy-app-theme);

**Comment by GarryBrown** [comment]
Yes, need replace all "md"  to "mat".
